### PR TITLE
[MultiDB] (except ./src and ./dockers dirs): replace redis-cli with sonic-db-cli and use new DBConnector

### PR DESCRIPTION
**Pull request by dzhangalibaba**
this commit made changes in sonic-buildimage module directory except ./dockers/(this is done in another PR) amd ./src/(will do in next PR)

* replace redis-cli with sonic-db-cli
* replace old DBConnector with new DBConnector
* vs tests passed, loaded on DUT, looks good
* check/compare  each output between redis-cli and sonic-db-cli in this PR, make sure they are the same format.

Signed-off-by: Dong Zhang d.zhang@alibaba-inc.com

### CODE REVIEW

**Comment by dzhangalibaba** [comment]
retest vsimage please

**Comment by dzhangalibaba** [comment]
retest vsimage please

**Comment by dzhangalibaba** [comment]
@qiluo-msft We can start review on this PR now

File: files/build_templates/docker_image_ctl.j2 [hunk 1]
```diff
@@ -113,7 +113,10 @@ function preStartAction()
     fi
 
     echo -n "" > /tmp/snmpagentaddr.yml
-    keys=`/usr/bin/redis-cli -n 4 keys "SNMP_AGENT_ADDRESS_CONFIG|*"`
+    # we should avoid using 'keys' operation via redis-cli or sonic-db-cli
```
Comment (qiluo-msft) [inline files/build_templates/docker_image_ctl.j2]: add explicit TODO as a marker in the comment.

Comment (dzhangalibaba) [reply files/build_templates/docker_image_ctl.j2]: updated the TODO marker, it trigger the testing again and you may need to approve it later again.

**Comment by dzhangalibaba** [comment]
retest baseimage please

**Comment by dzhangalibaba** [comment]
retest broadcom please

**Comment by dzhangalibaba** [comment]
retest vsimage please

**Comment by dzhangalibaba** [comment]
retest vsimage please

**Comment by qiluo-msft** [comment]
Since the previous commit passed PR checks, the new commit only modifies code comment, we are safe to merge.
```
@dzhangalibaba update comment d4b8601
```

**Pull request merged**